packages/devtools_app/lib/src/memory/flutter/memory_heap_tree_view.dart

```diff
@@ -263,23 +281,59 @@ class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
     );
   }
 
+  bool escapeHit = false;
+  TextEditingController textFieldController;
+  TextField searchField;
+
+  TextField createSearchField() {
+    // Creating new TextEditingController.
+    textFieldController = TextEditingController();
+
+    return TextField(
+      autofocus: true,
+      onChanged: (value) {
+        // TODO(terry): Extend search support beyond heat map.
+        if (controller.showHeatMap) {
+          controller.search = value;
+        }
+      },
+      controller: textFieldController,
+      decoration: const InputDecoration(
+        isDense: true,
+        border: OutlineInputBorder(),
+        labelText: 'Search',
+      ),
+    );
+  }
+
   Widget _buildSearchFilterControls() {
+    searchField = createSearchField();
+
+    final searchAndRawKeyboard = RawKeyboardListener(
+      child: searchField,
+      focusNode: FocusNode(),
```

Comment:
instead of creating a new focus node here, can you create it in a stateful widget so the lifecycle is a bit longer and you can call dispose on the focus node. 
that way selection won't be lost after for example doing a hot reload or otherwise rebuilding this portion of the widget tree.

Reply:
Done - good catch similar to what I did for TextField too.